compiler/src/yul/mappers/contracts.rs

```diff
@@ -2,46 +2,30 @@ use crate::errors::CompileError;
 use crate::yul::mappers::{
     constructor,
     functions,
-    types,
-};
-use crate::yul::namespace::events::Event;
-use crate::yul::namespace::scopes::{
-    ContractScope,
-    ModuleScope,
-    Scope,
-    Shared,
-};
-use crate::yul::namespace::types::{
-    FixedSize,
-    Type,
 };
 use crate::yul::runtime::abi as runtime_abi;
 use crate::yul::runtime::functions as runtime_functions;
 use fe_parser::ast as fe;
 use fe_parser::span::Spanned;
-use std::rc::Rc;
+use fe_semantics::Context;
 use yultsur::*;
 
 /// Builds a Yul object from a Fe contract.
 pub fn contract_def(
-    module_scope: Shared<ModuleScope>,
+    context: &Context,
     stmt: &Spanned<fe::ModuleStmt>,
 ) -> Result<yul::Object, CompileError> {
-    if let fe::ModuleStmt::ContractDef { name: _, body } = &stmt.node {
-        let contract_scope = ContractScope::new(module_scope);
-
+    if let (Some(functions), fe::ModuleStmt::ContractDef { name: _, body }) =
+        (context.get_contract(stmt), &stmt.node)
+    {
         let mut statements = body.iter().try_fold::<_, _, Result<_, CompileError>>(
             vec![],
             |mut statements, stmt| {
                 match &stmt.node {
-                    fe::ContractStmt::ContractField { .. } => {
-                        contract_field(Rc::clone(&contract_scope), stmt)?
-                    }
-                    fe::ContractStmt::EventDef { .. } => {
-                        event_def(Rc::clone(&contract_scope), stmt)?
-                    }
+                    fe::ContractStmt::ContractField { .. } => {}
+                    fe::ContractStmt::EventDef { .. } => {}
```

Comment:
These empty blocks look suspicious, did we lose existing functionality?

Reply:
Both contract fields and event definitions only modify scope, so there is nothing to map.

Modifications to scope happen [here](https://github.com/ethereum/fe/blob/598d4c03d3431c8681f3a4dc8dc274f79d50235d/semantics/src/traversal/contracts.rs#L55-L91) in the semantic pass.